### PR TITLE
[CP-1117] Implementing draft's functionality

### PR DESCRIPTION
**Pull request by igorbezsmertnyi**
Jira: [CP-1117]

**Description**
- Implementing the draft's funtionality

<details>
<summary><b>Screenshots</b></summary>
Screen Shot 2022-07-12 at 1 13 29 PM
</details>

[CP-1117]: https://appnroll.atlassian.net/browse/CP-1117?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/messages/selectors/get-thread-draft-messages.selector.test.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            firstDraftMessage.id,
+            firstDraftMessage.id,
+          ],
+        },
+      },
+    } as ReduxRootState
+    expect(getThreadDraftMessagesSelector(thread.id)(state)).toEqual(
+      firstDraftMessage
+    )
+  })
```
Comment (wojcik92michal) [inline packages/app/src/messages/selectors/get-thread-draft-messages.selector.test.ts:154]: Is it ok that there is such a mapping 
```
          [firstDraftMessage.id]: firstDraftMessage,
          [secondDraftMessage.id]: firstDraftMessage,
```
and
```
            outboxMessage.id,
            firstDraftMessage.id,
            firstDraftMessage.id,
```
? Shouldn't it be

```
          [firstDraftMessage.id]: firstDraftMessage,
          [secondDraftMessage.id]: secondDraftMessage,
```

and 

```
            outboxMessage.id,
            firstDraftMessage.id,
            secondDraftMessage.id,
```
?

File: packages/app/src/messages/selectors/get-thread-draft-messages.selector.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { createSelector } from "reselect"
+import { MessageType } from "App/messages/constants"
+import { Message } from "App/messages/dto"
+import { messagesStateSelector } from "App/messages/selectors/messages-state.selector"
+
+export const getThreadDraftMessagesSelector = (threadId: string) =>
```
Comment (wojcik92michal) [inline packages/app/src/messages/selectors/get-thread-draft-messages.selector.ts:11]: It returns one message, but the name indicates that it returns multiple.
Shouldn't it be then `getThreadDraftMessageSelector`?

File: packages/app/src/messages/selectors/get-thread-draft-messages.selector.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  createSelector(
+    messagesStateSelector,
+    ({ messageIdsInThreadMap, messageMap }) => {
+      const messageIds = messageIdsInThreadMap[threadId] ?? []
+      return messageIds
+        .map((messageId) => messageMap[messageId])
+        .find((message: Message) => message.messageType === MessageType.DRAFT)
```
Comment (wojcik92michal) [inline packages/app/src/messages/selectors/get-thread-draft-messages.selector.ts:18]: Not 100% sure if I get it right, but the test description says that `returns last draft message if many exists`, while `find` returns the value of the first element that passes a test. Is it ok?

File: packages/app/src/messages/selectors/get-thread-draft-messages.selector.test.ts (hunk 1 above)
Comment (igorbezsmertnyi) [reply packages/app/src/messages/selectors/get-thread-draft-messages.selector.test.ts:154]: @wojcik92michal I don't understand what exactly you mean here?

File: packages/app/src/messages/actions/update-message.action.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+export const updateMessage = createAsyncThunk<AppError | void, Message>(
+  MessagesEvent.UpdateMessage,
+  async (message, { rejectWithValue }) => {
+    const { status, error } = await updateMessageRequest(message)
+
+    if (error || status !== RequestResponseStatus.Ok) {
+      return rejectWithValue(
+        new AppError(
+          MessagesError.UpdateMessageError,
+          (error as unknown as Error)?.message || "Update failed"
```
[the remark this answers is not in the document]
Comment (igorbezsmertnyi) [reply packages/app/src/messages/actions/update-message.action.ts]: https://github.com/mudita/mudita-center/pull/835/commits/536160a8e559ce4417ad5e091ffd4bbf27e75d1d

File: packages/app/src/messages/presenters/thread.presenter.ts [hunk 5]
```diff
@@ -31,6 +30,13 @@ export class ThreadPresenter {
     }
   }
 
+  private static buildMessageSnippet(thread: PureThread): string {
+    return [
+      ...(thread.messageType === PureMessageType.DRAFT ? ["Draft"] : []),
```
[the remark this answers is not in the document]
Comment (igorbezsmertnyi) [reply packages/app/src/messages/presenters/thread.presenter.ts]: https://github.com/mudita/mudita-center/pull/835/commits/5046ca2ed7115e736d9a47fa73894145381a1848

File: packages/app/src/messages/selectors/get-thread-draft-messages.selector.ts (hunk 3 above)
Comment (igorbezsmertnyi) [reply packages/app/src/messages/selectors/get-thread-draft-messages.selector.ts]: https://github.com/mudita/mudita-center/pull/835/commits/31b4aae7dbdc63e21c41fb1e179a2ea58b9d9d98

File: packages/app/src/messages/selectors/get-thread-draft-messages.selector.ts (hunk 2 above)
Comment (igorbezsmertnyi) [reply packages/app/src/messages/selectors/get-thread-draft-messages.selector.ts]: https://github.com/mudita/mudita-center/pull/835/commits/31b4aae7dbdc63e21c41fb1e179a2ea58b9d9d98

File: packages/app/src/messages/selectors/get-thread-draft-messages.selector.test.ts (hunk 1 above)
Comment (igorbezsmertnyi) [reply packages/app/src/messages/selectors/get-thread-draft-messages.selector.test.ts]: https://github.com/mudita/mudita-center/pull/835/commits/31b4aae7dbdc63e21c41fb1e179a2ea58b9d9d98

File: packages/app/src/messages/selectors/get-messages-by-thread-id.selector.ts [hunk 6]
```diff
@@ -21,7 +22,9 @@ export const getMessagesByThreadIdSelector = (
     messagesStateSelector,
     ({ messageIdsInThreadMap, messageMap }) => {
       const messageIds = messageIdsInThreadMap[threadId] ?? []
-      const messages = messageIds.map((messageId) => messageMap[messageId])
+      const messages = messageIds
+        .map((messageId) => messageMap[messageId])
+        .filter((message: Message) => message.messageType !== MessageType.DRAFT)
```
[the remark this answers is not in the document]
Comment (igorbezsmertnyi) [reply packages/app/src/messages/selectors/get-messages-by-thread-id.selector.ts:27]: https://github.com/mudita/mudita-center/pull/835/commits/31b4aae7dbdc63e21c41fb1e179a2ea58b9d9d98

**Review by wojcik92michal — DISMISSED** [review]

**Review by wercix — CHANGES_REQUESTED** [review]
There is `: `  in thread snippet if message is not a draft
Screen Shot 2022-07-13 at 13 46 44

File: packages/app/src/messages/presenters/thread.presenter.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (wojcik92michal) [reply packages/app/src/messages/presenters/thread.presenter.ts]: Sure

[the remark this answers is not in the document]
Comment (igorbezsmertnyi) [reply packages/app/src/messages/presenters/thread.presenter.ts:35]: https://github.com/mudita/mudita-center/pull/835/commits/cfa5c478cc61143d8d22426b6fa9877df7c1119f

**Review by wercix — APPROVED** [review]

**Pull request merged**